modules/library/metadata/src/main/java/org/geotools/util/Utilities.java

```diff
@@ -670,4 +678,100 @@ public static void ensureNonNull(final String name, final Object object)
         }
     }
 
+    /**
+     * Makes sure that an argument is non-null and throws an {@link IllegalArgumentException}
+     * if it is.
+     *
+     * @param name argument name
+     * @param object argument
+     * @throws IllegalArgumentException if {@code object} is null.
+     */
+    public static void ensureArgumentNonNull(final String name, final Object object)
+            throws IllegalArgumentException {
+        if (object == null) {
+            throw new IllegalArgumentException(Errors.format(ErrorKeys.NULL_ARGUMENT_$1, name));
+        }
+    }
+
+    /**
+     * Creates a stream over the specified {@link Iterable}'s elements.
+     *
+     * This operation is lazy in the sense that it does not iterate over the elements.
+     *
+     * @param iterable the iterable to stream over
+     * @param <T> the type of elements contained in the iterable
+     * @return a stream over the iterable's elements
+     */
+    public static <T> Stream<T> stream(Iterable<T> iterable) {
+        return StreamSupport.stream(iterable.spliterator(), false);
+    }
+
+    /**
+     * Creates a stream over the specified {@link Iterator}'s elements.
+     *
+     * This operation is lazy in the sense that it does not iterate over the elements.
```

Comment:
Change this to "does not iterate over the elements until the Stream is consumed"